decisions/backlog/0022-oauth-token-endpoint.md

```diff
@@ -0,0 +1,71 @@
+# 22. Oauth2.0 Token endpoint url (Proposal #1)
+
+Date: 2023-09-19
+
+## Status
+
+In Progress
+
+## Context
+
+The Tech. Specs. (both V1 and V2) specify that the token endpoint for the Action `Authenticate` must conform to the following structure: `AuthSubPath/auth/token`. This specification goes beyond what is suggested in the referenced standard 'The OAuth 2.0 Authorization Framework' (https://www.rfc-editor.org/rfc/rfc6749), which specifies the path to have the structure as: `AuthSubPath/token`. This more strict specification of the path creates conformance challenges for host systems that:
+    
+* reuse an exsting service that is implemented based on the standard (rfc-6749), but does not offer the capability to control the structure of the token endpoint in full.
+    
+For example, the following path `https://some-domain.com/oauth/token` adheres to the standard (rfc-6749) but is not conforming to the current Tech. Specs. (see 6.5.1). The same would apply to the path `https://some-domain.com/token`.
+
+## Summary
+
+We propose to change the specified structure of the token endpoint for the Action `Authenticate` in the Tech. Specs. The current structure:
+
+* POST `AuthSubPath/auth/token`
+
+The proposed structure:
+
+* POST `AuthSubPath/token`
+
+This change aligns the Tech. Specs. to the standard (rfc-6749) and grants host systems more flexibility for implementing their token endpoint. The proposed change does not impact the authentication flow beyond the path. 
+
+As the proposed change 'loosens' the current specification, backwards-compatibility is ensured. Any host system that is considered to have a conformant implementation of the Action `Authenticate` under the current Tech. Specs. will also be conformant with the proposed change. Logically, any path that adheres to the currently specified structure `AuthSubPath/auth/token` will also adhere to the proposed structure `AuthSubPath/token`.
```

Comment:
Backwards-compatibility is certainly ensured from the point of view of **Data Owners** (because `AuthSubPath/auth/token` is also `AuthSubPath/token`), but is unclear to me whether that is also the case from the point of view of **Data Recipients**. 

If a Data Recipient on 2.0.1 tries to get a token from a solution with this update, they will call the `AuthSubPath/auth/token` without any guarantee that it will work nor a way of knowing which endpoint they should use instead.

Let me try to clarify what I mean by an example:
`Solution A` follows this update and implements `AuthSubPath/token`
`Solution B` is on 2.0.1 and, learning about `Solution A` decides to call their authentication endpoint. Since they are in 2.0.1, they will assume that they can call `AuthSubPath/auth/token`. They might be surprised that it doesn’t work and won’t have a way of finding out what the correct endpoint is.